File: algorithms/2_medium/K/00650_2_keys_keyboard/solution.c
static inline int largest_proper_divisor(const int n)
{
    const int n_sqrt = (int) sqrt(n); 
    for (int i = 2; i <= n_sqrt; i += 1) {
        if (n % i == 0) {
            return n / i;
        }
    }

    return 1;
}

int minSteps(int n)
{
    assert(n > 0);
    int cnt = 0;

    while (n > 1) {
        const int lpd = largest_proper_divisor(n);
        cnt += n / lpd;
        n = lpd;
    }

    return cnt;
}

File: algorithms/2_medium/K/00650_2_keys_keyboard/solution.rs
impl Solution {
    fn largest_proper_divisor(n: i32) -> i32 {
        for i in 2..=n.isqrt() {
            if n % i == 0 {
                return n / i;
            }
        }
        1
    }

    pub fn min_steps(mut n: i32) -> i32 {
        debug_assert!(n > 0);
        let mut cnt = 0_i32;

        while n > 1 {
            let lpd = Self::largest_proper_divisor(n);
            cnt += n / lpd;
            n = lpd;
        }

        cnt
    }
}
